Source/WebCore/style/StyleTreeResolver.h

```diff
@@ -160,22 +163,31 @@ class TreeResolver {
 
     bool hasUnresolvedAnchorPosition(const Element&) const;
 
-    struct QueryContainerState {
-        OptionSet<Change> changes;
-        DescendantsToResolve descendantsToResolve { DescendantsToResolve::None };
-        bool invalidated { false };
-    };
-
     CheckedRef<Document> m_document;
     std::unique_ptr<RenderStyle> m_documentElementStyle;
 
     Vector<Ref<Scope>, 4> m_scopeStack;
     Vector<Parent, 32> m_parentStack;
     bool m_didSeePendingStylesheet { false };
 
-    UncheckedKeyHashMap<Ref<Element>, std::optional<QueryContainerState>> m_queryContainerStates;
+    // For elements that require style/layout interleaving (anchor positioning and
+    // container queries), the first style pass will defer resolving descendants.
+    // This is resumed in subsequent style passes, when there are enough information
+    // to do so. This struct preserves internal resolution state when descendant
+    // resolution is deferred, so it could be restores when descendant resolution
+    // resumes.
+    struct StyleLayoutInterleavedElementState {
```

Comment:
I would leave out the word Style here as we are in Style namespace.

Reply:
I changed it to `DeferredDescendantResolutionState` which is a bit more relevant